### PR TITLE
Rebuild the benchmark with reproducible PA and DPD results

### PR DESCRIPTION
**Pull request by gaochangw**
## Summary

- replace the legacy benchmark scripts and intermediate experiment artifacts with one provenance-checked PA/DPD benchmark workflow
- publish the latest APA_200MHz and DPA_160MHz report, figures, and machine-readable evidence
- stabilize GMP PA identification with column-scaled truncated SVD and explicit solver/rank provenance
- align CLI and Python API training defaults with the validated OpenDPDv2 recipe
- add TRes-GRU plus the optimized TRes-DeltaGRU execution paths and their regression coverage
- propagate PA delta thresholds, use the dataset sample rate for EVM, and warn before architecture-only model IDs overwrite prior artifacts

## Why

The previous report contained an unreproducible `-43.52 dB` APA GRU PA result and used an undertrained PA recipe. APA GMP was also numerically unstable because CUDA `gels` assumes a full-rank design matrix, while the 1,350-column GMP basis is ill-conditioned/rank-deficient.

The new workflow archives exact checkpoints/logs, records commands and source hashes, validates all evidence before rendering, and keeps only user-facing benchmark artifacts in the repository.

## Latest benchmark highlights

- APA_200MHz GMP PA: `-38.7020 / -38.6606 dB` validation/test NMSE, rank `650/1350`
- DPA_160MHz GMP PA: `-38.9088 / -39.4014 dB`, rank `918/1350`
- APA_200MHz GRU-H28 PA: `-38.8504 / -38.9367 dB`
- reproduction run `20260726T161245Z`: 24/24 jobs successful, 22 published results
- 62/62 source-snapshot members and 42 archived training artifacts hash-verified

## User/developer impact

Users can reproduce the complete report with:

```bash
bash benchmark/reproduce_benchmark_report.sh --device 0
```

The generated checkpoint, history, commands, job ledger, raw polynomial outputs, source snapshot, report, and figures remain bound to the timestamped run. Current defaults now use batch size 64, 300 epochs, initial learning rate `5e-3`, scheduler patience 5, and minimum learning rate `5e-5`.

Addresses #15.

## Validation

- `.venv/bin/pytest -q`: 216 passed, 1 PyTorch CUDA-context warning
- `.venv/bin/pytest -q tests/test_benchmark_regressions.py`: 26 passed
- `bash -n benchmark/reproduce_benchmark_report.sh`
- `git diff --check`
- full fresh benchmark reproduction and artifact/hash audit

### CODE REVIEW

**Review by gaochangw — COMMENTED** [review]
Self-review completed against the final net diff. I reviewed the benchmark runner and provenance validation, MP/GMP solver changes, PA/DPD model paths (including TRes-GRU and TRes-DeltaGRU), CUDA/Triton fast-path fallbacks, public API/default hyperparameters, benchmark-folder cleanup, and the new regression tests. I found no blocking issues. All six required CI checks pass across lint, package build, and Python 3.10-3.13. Squash merge is appropriate because the branch contains iterative benchmark commits whose intermediate artifacts intentionally net out of the final PR.
